src/reanimated2/animation/util.ts

```diff
@@ -49,23 +49,25 @@ interface RecognizedPrefixSuffix {
   strippedValue: number;
 }
 
-function recognizePrefixSuffix(value: string | number): RecognizedPrefixSuffix {
+export function recognizePrefixSuffix(
+  value: string | number
+): RecognizedPrefixSuffix {
   'worklet';
-  if (typeof value === 'string') {
-    const match = value.match(
-      /([A-Za-z]*)(-?\d*\.?\d*)([eE][-+]?[0-9]+)?([A-Za-z%]*)/
-    );
-    if (!match) {
-      throw new Error("[Reanimated] Couldn't parse animation value.");
-    }
-    const prefix = match[1];
-    const suffix = match[4];
-    // number with scientific notation
-    const number = match[2] + (match[3] ?? '');
-    return { prefix, suffix, strippedValue: parseFloat(number) };
-  } else {
+  if (typeof value === 'number') {
```

Comment:
I've moved this up, to make this less nested.
Do we need `prefix`?

Reply:
> I've moved this up, to make this less nested.
but why? 🤔 You significantly changed the logic of this method.

Reply:
The logic isn't changed drastically, since value is typed to be either a 'number' or 'string'. 
> Why?

Many devs encourage to prefer the less nested solution, since it is considered simpler. https://testing.googleblog.com/2017/06/code-health-reduce-nesting-reduce.html

Reply:
The logic isn't changed drastically, since value is typed to be either a 'number' or 'string'. 
> Why?

Many devs encourage to prefer the less nested solution, since it is considered simpler. https://testing.googleblog.com/2017/06/code-health-reduce-nesting-reduce.html

Reply:
But what about for example this cast?
https://github.com/software-mansion/react-native-reanimated/blob/4d4f2118553e330829b2d49d4ca3b4d92d3a8a55/src/reanimated2/animation/util.ts#L160